### DISCUSSION TITLE
The use of "default_class_id" in multiclass classification

### DISCUSSION THREAD

**Discussion by greenhawktobias** [azavea/raster-vision#1516] [Q&A] [answered]
Hi,
I am doing a test with some images of mowed grass and uncut grass on drone images using raster vision Release 0.13.0
I was thinking about the class ids to use.
I have defined the classes in my geojson label file (image)
Background is set to class_id=2, but **is it correct to set `default_class_id=0` in this case?**

**Another question:** Does chipsize affect the classification results? As in this case with a images resolution of 2 cm
and mowed grass area of 10*8 meters maybe smaller chipsize is beneficial?
 
```
CLASS_NAMES = [
    'mowed', 'uncut','background'
]
CLASS_COLORS = [
    '#ffff00', '#0000ff', '#00ffff'
```
`default_class_id=0`

` rasterizer_config=RasterizerConfig(background_class_id=2)))`

![classes](https://user-images.githubusercontent.com/79131575/198249734-6c861a64-4224-490f-b1b8-beadfb212203.JPG)
![scene2](https://user-images.githubusercontent.com/79131575/198254607-2d877c56-aa1a-42c0-93a1-56aaa313d72c.jpg)

**Comment by AdeelH** [comment, marked as answer]
`default_class_id` is for GeoJSON features that don't have any class info. If all the features in your GeoJSON have a class ID or name, then `default_class_id` will have no effect. If you set it to `None`, features without any class info will be dropped.

See [docs](https://docs.rastervision.io/en/latest/api_reference/_generated/rastervision.core.data.vector_transformer.class_inference_transformer.ClassInferenceTransformer.html#rastervision.core.data.vector_transformer.class_inference_transformer.ClassInferenceTransformer). These docs are not for v0.13 but the class inference logic is the same.

`default_class_id` is mostly useful when you have just one foreground class and all the GeoJSON features belong to that class but don't have class info in their properties. In such a case, `default_class_id` saves you the trouble of altering your GeoJSON file,
